client/src/helpers/fhir/fhir.js

```diff
@@ -90,6 +94,37 @@ export const getObservationValue = (obs, defaultValue) => (
   get(obs, 'valueString', get(obs, 'note[0].text', defaultValue))
 )
 
+export const getNoteComment = (serviceRequest) => {
+  const value = serviceRequest?.note?.[0]?.text;
+  return value != EMPTY_NOTE_TEXT ? value : null
+}
+export const getNoteStatus = (serviceRequest) => {
+  const value = serviceRequest?.note?.[1]?.text;
+  return value != EMPTY_NOTE_TEXT ? value : null
+}
+
+export const updateNoteComment = (notes, comment) => {
+  if (Array.isArray(notes)) {
```

Comment:
Why checking if it's an array? Shouldn't it always be an array?

Reply:
Don't see anything wrong with adding some robustnes, anyway will be removed if I put all theses functions inside a typescript file as you proposed.

Reply:
Indeed there is nothing wrong with it. It is good code. If you wanna leave it you can.

Just would like to say why I think, in this specific case (again,it's on case-by-case basis),  I believe it's better not make that check. In theory, we expect an array. So, if we pass by error another type of variable (it's easy to to do in Javascript) it may hide a bug for a longer time than it needs too. Plus, it adds some "noise/complexity".